extension/src/popup/helpers/soroban.ts

```diff
@@ -121,18 +128,12 @@ export const getArgsForTokenInvocation = (
 
   switch (fnName) {
     case SorobanTokenInterface.transfer:
-      from = StrKey.encodeEd25519PublicKey(
-        args[0].address().accountId().ed25519(),
-      );
-      to = StrKey.encodeEd25519PublicKey(
-        args[1].address().accountId().ed25519(),
-      );
+      from = addressToString(args[0].address());
```

Comment:
Now that we support transfers to C addresses, I ran into this where from and to are actually not always `accountId` and can also be `contractId`.